File: src/main/java/com/GoldenMine/parser/postposition/IPostPosition부사어로.java
package com.GoldenMine.parser.postposition;

public class IPostPosition부사어로 implements IPostPosition {
    @Override
    public String getType() {
        return "부사어로";
    }

    @Override
    public int verify(CharSequence last) {
        if(last.length() >= 2) {
            char ch = last.charAt(last.length() - 2);
            char ch2 = last.charAt(last.length() - 1);

            return (ch == '으' && ch2 == '로') ? 2 : ((ch2 == '로') ? 1 : -1);
        } else if(last.length() >= 1){
            char ch2 = last.charAt(last.length() - 1);

            return  ((ch2 == '로') ? 1 : -1);
        } else {
            return -1;
        }
    }

    @Override
    public JosaCommunity getJosaCommunity() {
        return JosaCommunity.격조사;
    }
}

File: src/main/java/kr/goldenmine/parser/postposition/IPostPosition부사어로.kt
package kr.goldenmine.parser.postposition

class IPostPosition부사어로 : IPostPosition {
    override val type: String
        get() = "부사어로"

    override val josaCommunity: JosaCommunity
        get() = JosaCommunity.격조사

    override fun verify(last: CharSequence): Int {
        if (last.length >= 2) {
            val ch = last[last.length - 2]
            val ch2 = last[last.length - 1]

            return if (ch == '으' && ch2 == '로') 2 else if (ch2 == '로') 1 else -1
        } else if (last.length >= 1) {
            val ch2 = last[last.length - 1]

            return if (ch2 == '로') 1 else -1
        } else {
            return -1
        }
    }
}
